src/cryptoadvance/specter/server_endpoints/wallets_api.py

```diff
@@ -98,6 +100,50 @@ def fees_old(blocks):
     return app.specter.estimatesmartfee(int(blocks))
 
 
+@wallets_endpoint_api.route("/get_new_notifications", methods=["GET"])
+@login_required
+def get_new_notifications():
+    """
+    Gets all notifications meant for the browser to display in a dict of structure:
+        {
+            "WebAPI": [notification1, notification2, ...],
+        }
+    """
+
+    def myjsonconverter(o):
+        if isinstance(o, datetime):
+            return o.timestamp()
+
+    js_notifications_dict = {}
+    for (
+        ui_notification
+    ) in app.specter.user_manager.get_user().notification_manager.ui_notifications:
+        if ui_notification.name in {"WebAPI", "js_message_box", "js_console"}:
+            notifications = ui_notification.read_and_clear_js_notification_buffer()
+            if notifications:
+                js_notifications_dict[ui_notification.name] = notifications
+
+    return json.dumps(js_notifications_dict, default=myjsonconverter)
+
+
+@wallets_endpoint_api.route("/create_notification", methods=["POST"])
+@login_required
+def create_notification():
```

Comment:
I added a docstring and the give an error if the "title" is missing. Can you check again if this is OK?

Reply:
The boilerplate code (transforming data to get it ready for the next level) is still not very convenient and also doesn't scale. I thought about it and the way to avoid such kind of boilerplate is to send json in the first place. I did implement that roughly:
https://github.com/relativisticelectron/specter-desktop/compare/20220616_notification_system...k9ert:my_notification_system

However, in that way we loose CSRF-Protection and i don't think this is acceptable and i couldn't figure out whether there is a way to have CSRF protection with json-requests. Probably that's not possible easily.

So in order to have at least a meaningful boilerplate code, i'd suggest to use the `request.form.get("title")` form and build your dict that way or make the calls more static through the whole stack. We don't have any tests yet and this kind of flexible method calling makes it really difficult to track down issues if something goes wrong.

Reply:
I will take over some ideas from your [code](https://github.com/relativisticelectron/specter-desktop/compare/20220616_notification_system...k9ert:my_notification_system)  and try to incorporate it (while keeping CSRF)